### ISSUE TITLE
Configurations for DeepLab on Cityscapes

### ISSUE THREAD

**Issue by WilhelmT** [yassouali/pytorch_segmentation#54]
Hello. Thank you for your pleasant implementation!

I have been running multiple experiments with DeepLab v3 (Resnet101 backbone) on the Cityscapes dataset, and have been consistently getting at most 67-70 MIoU, while I believe it should be around 80. I am training on full resolution images with crop size 768, otherwise my settings is identical to your PSPNet setup. Do you have any suggestions for how to set my configuration in this setting?

**Comment by WilhelmT** [comment]
There's a similar implementation to yours, https://github.com/CoinCheung/DeepLab-v3-plus-cityscapes, which yields 80 MIoU. Some differences I've observed are that you have drop-out layers and no Ohem-loss. I have not tried to add the Ohem-loss but when I removed the dropout-layers from the Deeplabv3plus, the results deteriorated even further (even on the training data which seems extraordinarily weird). I will send you my full config file tomorrow!

**Comment by yassouali** [comment]
Thanks for the raising the issue, I will try to find the issue.

**Comment by deepakksingh** [comment]
> There's a similar implementation to yours, https://github.com/CoinCheung/DeepLab-v3-plus-cityscapes, which yields 80 MIoU. Some differences I've observed are that you have drop-out layers and no Ohem-loss. I have not tried to add the Ohem-loss but when I removed the dropout-layers from the Deeplabv3plus, the results deteriorated even further (even on the training data which seems extraordinarily weird). I will send you my full config file tomorrow!

Hey, I was wondering if you could share your config file for deeplab on cityscapes.
Even I'm trying to reproduce the results but I'm stuck with 64.5 mIoU

**Comment by xuyanr** [comment]
I want the config file for deeplab on cityscapes, too. If it's  convenient for you.

**Comment by Leo-cloud** [comment]
Hello. Thank you for your pleasant implementation!
I meet some problems when I run the train.py , can you tell me how to solve it ? thank you!
![2020-05-19 14-16-25屏幕截图](https://user-images.githubusercontent.com/54903022/82291523-b6ae8680-99db-11ea-8550-344fda1e4cfd.png)

**Comment by yassouali** [comment]
@Leo-cloud make sure the name of the dataset is correct (the same one as the class names in \dataloaders\__init__.py)

**Comment by Leo-cloud** [comment]
> @Leo-cloud make sure the name of the dataset is correct (the same one as the class names in `\dataloaders\__init__.py`)

Thank you so much ! But now I meet some other problems, it is diffcult for me to change config file, can you share your config file for deeplab on cityscapes. thank you very much.
![2020-05-19 14-16-25屏幕截图](https://user-images.githubusercontent.com/54903022/82310770-604f4100-99f7-11ea-8cf0-c67513e65540.png)

**Comment by yassouali** [comment]
I think the error come from the split used, in cityscapes, there is only `train` and  `val`, so simply replace `"split": "train_aug"` with `"split": "train"`.

**Comment by Leo-cloud** [comment]
> I think the error come from the split used, in cityscapes, there is only `train` and `val`, so simply replace `"split": "train_aug"` with `"split": "train"`.

it can be run now, but it needs too much CUDA memory，how to solve it？tahnk you! Should I reduce the size of dataset or the "batch_size".
![2020-05-20 17-38-02屏幕截图](https://user-images.githubusercontent.com/54903022/82431058-c18f1700-9ac0-11ea-9bb8-ad08ecd3f9ce.png)

**Comment by yassouali** [comment]
It is your choice, you can either reduce the size of the images of the batch size, imo better to reduce the batch size.

**Comment by Leo-cloud** [comment]
> It is your choice, you can either reduce the size of the images of the batch size, imo better to reduce the batch size.
Thank you for your reply. I have finished the training, but when I want to segment some pictures, some erros occurred.
![image](https://user-images.githubusercontent.com/54903022/82516394-2bed9900-9b4d-11ea-8c06-a14520f0232e.png)

**Comment by yassouali** [comment]
I think no images were found in the folder, try printing the folder files to make sure you have the right format / folder (you might need to change the extension if the images are png and not jpeg).

**Comment by Leo-cloud** [comment]
> I think no images were found in the folder, try printing the folder files to make sure you have the right format / folder (you might need to change the extension if the images are png and not jpeg).

I put both png and jpeg images in the folder, but the error still exists.

**Comment by Leo-cloud** [comment]
> I think no images were found in the folder, try printing the folder files to make sure you have the right format / folder (you might need to change the extension if the images are png and not jpeg).

Inference can be run when  I change the the image to jpg, but it have a RuntimeError, why does it happen ? Does it need more CUDA memory than traning ? How to solve it?
![图片](https://user-images.githubusercontent.com/54903022/82555107-ba880780-9b99-11ea-86e0-7bce07d1a8a1.png)

**Comment by yassouali** [comment]
Generally, not, the inference is a lot less memory expensive, but given that here you use the original image sizes, this might be the case, you can apply a simple resize of the images after reading them (after line 132, `image = image.resize((new_width, new_height), Image.BICUBIC)`)

**Comment by Leo-cloud** [comment]
> Generally, not, the inference is a lot less memory expensive, but given that here you use the original image sizes, this might be the case, you can apply a simple resize of the images after reading them (after line 132, `image = image.resize((new_width, new_height), Image.BICUBIC)`)

How can I get only a certain kind of segmentation image in the original image, such as the cars,  people or  buildings？

**Comment by Leo-cloud** [comment]
> Hello. Thank you for your pleasant implementation!
> 
> I have been running multiple experiments with DeepLab v3 (Resnet101 backbone) on the Cityscapes dataset, and have been consistently getting at most 67-70 MIoU, while I believe it should be around 80. I am training on full resolution images with crop size 768, otherwise my settings is identical to your PSPNet setup. Do you have any suggestions for how to set my configuration in this setting?

Can you share your Configurations for DeepLab on Cityscapes? I only get 49% mIoU, thank you very much.

**Comment by yassouali** [comment]
continues on https://github.com/yassouali/pytorch_segmentation/issues/54
